src/applications/vaos/appointment-list/redux/reducer.js

```diff
@@ -150,22 +152,18 @@ export default function appointmentsReducer(state = initialState, action) {
         facilityData,
       };
     }
-    case FETCH_REQUEST_DETAILS: {
+    case (FETCH_REQUEST_DETAILS, FETCH_CONFIRMED_DETAILS): {
       return {
         ...state,
         appointmentDetailsStatus: FETCH_STATUS.loading,
-        currentAppointment: null,
       };
     }
-    case FETCH_REQUEST_DETAILS_SUCCEEDED: {
-      const appointmentDetails = { ...state.appointmentDetails };
-
-      appointmentDetails[action.id] = action.request;
-
+    case (FETCH_REQUEST_DETAILS_SUCCEEDED, FETCH_CONFIRMED_DETAILS_SUCCEEDED): {
       return {
         ...state,
-        currentAppointment: action.request,
-        appointmentDetails,
+        appointmentDetails: {
```

Comment:
I think you need a `...state.appointmentDetails,` in this object, so that we don't remove any other appointment detail items that might be there.

Reply:
Fixed.